elastic/kibana
# [RAM] Move alerts filter controls to @kbn/alerts-ui-shared package

## Summary

Moves Security's alerts filter controls bar implementation to the `@kbn/alerts-ui-shared` package for re-use in the Stack alerts (unified) page and Observability solution (separate PR).

<img width="1339" alt="image" src="https://github.com/elastic/kibana/assets/18363145/986e984a-0290-4980-859b-9d6e02ef335f">

## To verify

1. Enable the Stack alerts page feature flag:

    ```yaml
    # config/kibana.dev.yml
    
    xpack.trigger_actions_ui.enableExperimental:
      - globalAlertsPage
    ```
2. Create some Stack/O11y rules that fire some alerts
3. Navigate to `Management > Stack management > Alerts`, you should see the new controls below the KQL bar.
4. Check that the filters apply correctly and reflect on the table results
5. Check that the filter controls can be customized (`••• > Edit controls`, add/remove controls, save)
6. Check that the customized configuration persists across reloads
7. Check that the filter controls can be temporarily overridden through the URL param `filterControls`
8. Check that when the filters are overridden through the URL, a notice is shown to the user with CTAs to discard/save the temporary filter configuration
---
1. Navigate to `Security > Alerts > Manage rules`
2. Create one or more security rule that fire alerts
3. Go back to the security alerts page
4. Repeat steps 4-8

Closes #176711
